classes/condition.php

```diff
@@ -23,163 +27,135 @@
 namespace availability_shibboleth2fa;
 
 use availability_shibboleth2fa\event\user_2fa_loggedin;
+use coding_exception;
+use context_course;
+use core\event\course_deleted;
+use core\event\user_deleted;
+use core\event\user_enrolment_deleted;
+use core_availability\condition as abstract_condition;
 use core_availability\info;
 use core_availability\info_module;
 use core_availability\info_section;
+use dml_exception;
+use html_writer;
+use moodle_exception;
+use moodle_url;
+use stdClass;
 
-defined('MOODLE_INTERNAL') || die();
+/**
+ * Availability condition class.
+ *
+ * @see https://moodledev.io/docs/4.4/apis/plugintypes/availability#classesconditionphp
+ *
+ * @package    availability_shibboleth2fa
+ * @copyright  2021 Lars Bonczek, innoCampus, TU Berlin
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class condition extends abstract_condition {
 
-class condition extends \core_availability\condition {
+    /** @var int|null cached ID of user with an exception */
+    private static int|null $exceptioncacheuser = null;
 
-    /** @var int userid */
-    private static $exceptioncacheuser = null;
-    /** @var array list of courses with exception */
-    private static $exceptioncache = [];
+    /** @var int[] cached list of course IDs with exceptions */
+    private static array $exceptioncache = [];
 
     /**
-     * Determines whether a particular item is currently available
-     * according to this availability condition.
-     *
-     * If implementations require a course or modinfo, they should use
-     * the get methods in $info.
-     *
-     * The $not option is potentially confusing. This option always indicates
-     * the 'real' value of NOT. For example, a condition inside a 'NOT AND'
-     * group will get this called with $not = true, but if you put another
-     * 'NOT OR' group inside the first group, then a condition inside that will
-     * be called with $not = false. We need to use the real values, rather than
-     * the more natural use of the current value at this point inside the tree,
-     * so that the information displayed to users makes sense.
+     * {@inheritDoc}
      *
      * @param bool $not Set true if we are inverting the condition
      * @param info $info Item we're checking
-     * @param bool $grabthelot Performance hint: if true, caches information
-     *   required for all course-modules, to make the front page and similar
-     *   pages work more quickly (works only for current user)
+     * @param bool $grabthelot Performance hint: if true, caches information required for all course-modules, to make the front page
+     *                         and similar pages work more quickly (works only for current user)
      * @param int $userid User ID to check availability for
-     * @return bool True if available
```

Comment:
You are right, this ended up missing because I first removed everything using `{@inheritDoc}` and only added the `@throws` tag that was missing in the base method. Then the CS complained that the parameters were missing in our PHPDoc, so I added them back in. Apparently the missing `@return` is fine for the code sniffer though. But I agree, if the parameters are explained (redundantly) the return should be too. I'll add it back.

Reply:
Fixed with 0953ee0.